File: app/src/main/java/com/example/home/network/RetroService.java
package com.example.home.network;

import com.example.home.model.Page;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RetroService {

    @GET("/api/v1/search_by_date/")
    Single<Page> getRemotePage(@Query("tags") String tags, @Query("page") int page);

}

File: app/src/main/java/com/example/assignment13_mvvm_kotlin/network/RetroService.kt
package com.example.assignment13_mvvm_kotlin.network

import com.example.assignment13_mvvm_kotlin.model.Page
import retrofit2.http.GET
import retrofit2.http.Query

interface RetroService {

    @GET("/api/v1/search_by_date/")
    suspend fun getRemotePage(@Query("tags") tags: String?, @Query("page") page: Int): Page
}